### PR TITLE
feat(contextual-navigation): apply new specs - INNO-1236

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

Update component by applying new [specifications](https://webgate.ec.europa.eu/CITnet/confluence/display/NEXTEUROPA/Contextual+navigation+-+Design+specifications)

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit fa7ab81bbae85b1010633d165b8e0e19d764b11a

https://deploy-preview-954--europa-component-library.netlify.com

File: src/systems/ec/implementations/react/packages/contextual-navigation/ContextualNavigation.jsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import classnames from 'classnames';
+
+import VanillaContextualNavigation from '@ecl/ec-component-contextual-navigation/ec-component-contextual-navigation';
+import Link from '@ecl/ec-react-component-link/Link';
+
+export default class ContextualNavigation extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {};
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/contextual-navigation/ContextualNavigation.jsx]: why do you initialize a state here?

File: src/systems/ec/implementations/react/packages/contextual-navigation/ContextualNavigation.jsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    );
+
+    return (
+      <nav {...props} className={classNames} ref={this.contextualNavigationRef}>
+        <div className="ecl-contextual-navigation__label">{label}</div>
+        <ul
+          className="ecl-contextual-navigation__list"
+          data-ecl-contextual-navigation-list
+        >
+          {items.map((item, key) => (
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/contextual-navigation/ContextualNavigation.jsx]: `index` is more comprehensible than `key`

File: src/systems/ec/implementations/react/packages/contextual-navigation/ContextualNavigation.jsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+          {items.length > 3 && itemMoreMarkup}
+        </ul>
+      </nav>
+    );
+  }
+}
+
+ContextualNavigation.propTypes = {
+  author: PropTypes.string,
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/contextual-navigation/ContextualNavigation.jsx]: `author` is not used

File: src/systems/ec/implementations/react/packages/contextual-navigation/ContextualNavigation.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  render() {
+    const { label, itemMore, items, className, ...props } = this.props;
+
+    const classNames = classnames(className, 'ecl-contextual-navigation');
+
+    const itemMoreMarkup = (
+      <li className="ecl-contextual-navigation__item ecl-contextual-navigation__item--more">
+        <Link
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/contextual-navigation/ContextualNavigation.jsx]: "More" is a button, not a link

**Review by yhuard — CHANGES_REQUESTED** [review]
Overall it looks good. I've noticed some small details that can be fixed

**Review by yhuard — CHANGES_REQUESTED** [review]
React should handle the DOM mutations

File: src/systems/ec/implementations/react/packages/contextual-navigation/stories/Index.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        const vanillaContextualNavigation = new VanillaContextualNavigation(
+          element
+        );
+        vanillaContextualNavigation.init();
+
+        // Return new context
+        return { vanillaContextualNavigation };
+      }}
+      beforeUnmount={context => {
+        if (context.vanillaMessage) {
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/contextual-navigation/stories/Index.jsx]: `vanillaMessage` -> `vanillaContextualNavigation`

**Review by yhuard — CHANGES_REQUESTED** [review]

File: src/systems/ec/implementations/react/packages/contextual-navigation/stories/Index.jsx (hunk 5 above)
Comment (emeryro) [reply src/systems/ec/implementations/react/packages/contextual-navigation/stories/Index.jsx]: indeed...

**Review by yhuard — APPROVED** [review]

**Pull request merged**